### PR TITLE
Friends frontend

### LINKED ISSUE #14: `%pals` integration
Quick-select feature for challenging your friends.

Frontend and backend work.

### PR DESCRIPTION
**Pull request by bonbud-macryg**
This PR is functional but the UI has some rough edges, I'd like to hold off on polishing that up until this code is merged with the outgoing challenges list in #7, so I can get the whole Challenges menu working as intended.

### CODE REVIEW

File: src/frontend/tsx/Challenges.tsx [hunk 1]
```diff
@@ -77,16 +79,45 @@ export function Challenges () {
     await pokeAction(urbit, challenge(who, side, description), onError, onSuccess)
   }
 
+  const findFriends = async () => {
+    const friendsScry: Object = await urbit.scry({ app: 'chess', path: '/friends' })
+    const stringified: string = JSON.stringify(friendsScry, null, 2)
+    const parsed = JSON.parse(stringified) as { friends: Array<Ship> }
+    setFriends(parsed.friends)
+  }
```
Comment (ashelkovnykov) [inline src/frontend/tsx/Challenges.tsx]: This function should go in `ts/helpers/urbitChess` - that's the default place for pokes/scries

File: src/frontend/tsx/Challenges.tsx [hunk 2]
```diff
@@ -77,16 +79,45 @@ export function Challenges () {
...
+      resetChallengeInterface()
+    }
+    // XX: fill in a default description,
+    //     maybe current date like '~2022.12.7'
+    //
+    // XX: add this challenge to outgoingChallenges,
+    //     handle UI to show we can't challenge someone
+    //     who we've already challenged
+    await pokeAction(urbit, challenge('~' + opponent, Side.Random, ''), onError, onSuccess)
+  }
```
Comment (ashelkovnykov) [inline src/frontend/tsx/Challenges.tsx]: Would it be difficult to instead have the quick challenge button open the same pop-up that "New Challenge" opens, but pre-populate it with the friendly ship's `@p`? Would solve the problems of only being able to challenge as random, needing a default description, and replicating success/failure logic.

**Review by ashelkovnykov — CHANGES_REQUESTED** [review]
I'm not even going to test the interface myself, that's how much I trust you (totally not because I'm too lazy to spin up two moons and install `%chess` + `%pals` on both of them - how could you even think that)

File: src/frontend/tsx/Challenges.tsx [hunk 3]
```diff
@@ -120,16 +127,32 @@ export function Challenges () {
...
+                <div className='challenge-box'>
+                  <div className='row'>
+                    <div className='col'>
+                      <p className='friend'>~{friend}</p>
+                      {/* XX: win/loss history with this friend */}
+                      <p className='score'>0-0</p>
+                    </div>
+                  </div>
+                  <div className='col'>
+                    <button className='quick-game' onClick={() => {setWho('~' + friend); openModal() }}>Challenge</button>
```
Comment (rovmug-ticfyn) [inline src/frontend/tsx/Challenges.tsx]: Space needed after `{`
`...{ setWho('~' + friend)...`

**Comment by bonbud-macryg** [comment]
I think this PR's reformatted scry in `helpers/urbitChess` establishes a pattern which #6 could follow for getting the list of completed games. @brbenji

**Comment by brbenji** [comment]
> I think this PR's reformatted scry in `helpers/urbitChess` establishes a pattern which #6 could follow for getting the list of completed games. @brbenji

Cool! Soon I’ll be focusing in that direction.

File: src/frontend/tsx/Challenges.tsx [hunk 4]
```diff
@@ -186,8 +164,10 @@ export function Challenges () {
               className={(badChallengeAttempts > 0) ? 'rejected' : ''}
               type="text"
               placeholder={'~sampel-palnet'}
+              value={who}
               onChange={(e) => setWho(e.target.value)}
-              key={badChallengeAttempts}/>
+              key={badChallengeAttempts}
+              disabled={ challengingFriend ? true : false }/>
```
Comment (ashelkovnykov) [inline src/frontend/tsx/Challenges.tsx]: `challengingFriend` is already a boolean, don't need to use the ternary operator

File: src/frontend/tsx/Challenges.tsx [hunk 5]
```diff
@@ -1,6 +1,6 @@
 import React, { useState } from 'react'
 import Popup from 'reactjs-popup'
-import { pokeAction, challenge, acceptGame, declineGame } from '../ts/helpers/urbitChess'
+import { pokeAction, scry, challenge, acceptGame, declineGame, findFriends } from '../ts/helpers/urbitChess'
```
Comment (ashelkovnykov) [inline src/frontend/tsx/Challenges.tsx]: I think `scry` is unused

**Review by ashelkovnykov — APPROVED** [review]
LGTM - two minor comments, crush to one commit, and then feel free to merge yourself

**Pull request merged**